storage/store.go

```diff
@@ -144,6 +144,39 @@ func verifyKeys(start, end proto.Key, checkEndKey bool) error {
 	return nil
 }
 
+// verifyTransactionKeys verifies that requests that update transaction
+// keys have valid request keys matching with transaction keys.
+//
+// Request keys need to be same as transaction keys to guarantee that
+// at most one update command runs for a given transaction. (The command
+// queue ensures that.)
+func verifyTransactionKeys(args proto.Request) error {
+	header := args.Header()
+	if !proto.IsTransactionKeyUpdate(args) {
+		return nil
+	}
+	var txn *proto.Transaction
+	switch tArgs := args.(type) {
```

Comment:
just a heads up that `ResolveIntent` will no longer use the `Header.Txn` field in the near future, so it'll always have to be checked in the actual command.